docs/proposals/2437-volcano-scheduler/README.md

```diff
@@ -0,0 +1,251 @@
+
+First I will create my Queue in the cluster. The custom Queue may look as follows:
+
+```yaml
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: Queue
+metadata:
+  name: high-priority-queue
+spec:
+  weight: 1
+  reclaimable: false
+  capability:
+    cpu: 2
+```
+
+Then I specify the Queue name in ClusterTrainingRuntime spec:
+
+```yaml
+podGroupPolicy:
+    volcano:
+      queue: high-priority-queue
+```
+
+
+## Design Details
+
+As shown in the workflow diagram above, we decide to implement a runtime plugin for Volcano with the Kubeflow Trainer Pipeline Framework. It will:
+
+- **Build PodGroups** based on the *Training Runtime* configuration and calculate resource limits (e.g., `MinResource`).
+- **Manage PodGroups**
+   - Update: Update PodGroups and perform rescheduling when there are changes in cluster resource demands (e.g., changes in `LimitRange`).
+   - Suspended/Resumed: Support scheduling for suspended and resumed training jobs, with special handling of suspended jobs to ensure no new pods are started. (TrainJob may be set to suspend in its configuration or manually paused by the user.)
+- **Binding**: Bind PodGroups to TrainJobs, with their life cycle controlled by the TrainJob. For example, when a TrainJob is deleted, the associated PodGroup is also deleted.
+- **Apply PodGroups to the Cluster**: Submit PodGroup resources and associated scheduling configurations, allowing Volcano to manage pod scheduling.
+
+Note: The plugin is responsible only for configuring scheduling parameters, building and managing PodGroups. The actual scheduling management is handled by external schedulers (**volcano-controller**).
+
+### Volcano Scheduling API
+
+Currently, scheduling strategy parameters are set in the `PodGroupPolicy` of the `TrainingRuntimeSpec`. We introduce a new configuration struct, `VolcanoPodPolicySource`, which extends the existing `PodGroupPolicySource`:
+
+```golang
+// Only one of its members may be specified.
+type PodGroupPolicySource struct {
+        // Coscheduling plugin from the Kubernetes scheduler-plugins for gang-scheduling.
+	Coscheduling *CoschedulingPodGroupPolicySource `json:"coscheduling,omitempty"`
+
+	// Volcano plugin from the Volcano scheduler for gang-scheduling and advanced queue-based scheduling.
+	Volcano      *VolcanoPodPolicySource      `json:"volcano,omitempty"`
+}
+
+// VolcanoPodPolicySource configures scheduling behavior for Volcano.
+type VolcanoPodPolicySource struct {
+        // Queue name in Volcano. Defaults to "default" queue with the lowest weight.
+	Queue *string `json:"queue,omitempty"`
+
+        // PriorityClassName sets PodGroup priority. Optional.
+        // "system-node-critical" and "system-cluster-critical" are special keywords with the highest priorities.
+        PriorityClassName *string `json:"priorityClassName,omitempty"`
+}
+
```

Comment:
Can you close this code block to appropriately format the KEP ?

https://github.com/kubeflow/trainer/blob/0980c574058c48d79b423189af3c5c09766cea84/docs/proposals/2437-volcano-scheduler/README.md#volcano-scheduling-api